# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/muhammadimran5016/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

## 1) My lane (or freestyle) and why

**Lane:** Refresh / Content Opportunity Scoring

I chose this lane because it focuses on identifying which pages may be worth reviewing, refreshing, expanding, protecting, pruning, or monitoring. This connects the available search and content data to a practical decision: which pages should be prioritized for review. I want to investigate whether the starter dataset contains useful signals that can help prioritize content opportunities.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2) The question: decision, action, cost of a wrong call

**Search question:** Which pages in the dataset should be prioritized for content review or refresh?

**Decision:** Decide which pages should be reviewed first based on their available search and content signals.

**Action:** An SEO or content team could review the prioritized pages and decide whether to refresh, expand, protect, prune, or monitor them.

**Unit of analysis:** A page.

**Output:** A prioritized list or score showing which pages appear to have higher content refresh opportunities.

**Cost of a wrong recommendation:** A wrong recommendation could cause the team to spend time and resources refreshing a page that does not need it, while a page that actually needs attention may be overlooked.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import os
import sys
import subprocess

REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)

print("Working directory:", os.getcwd())
print("Starter data found:", os.path.exists("data/raw/content_refresh_anonymized.csv"))


Working directory: /content/flyrank-ml-internship-starter/flyrank-ml-internship-starter
Starter data found: True


In [9]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(), 4))
print("Average position:", round(df["avg_position"].mean(), 2))

Number of pages: 30000
Average CTR: 0.5107
Average position: 16.34


## 3) Quick look at the data

The starter dataset contains **30,000 pages**. The average CTR is **0.5107**, and the average position is **16.34**. These numbers show that the dataset has a large number of pages and measurable search-performance signals that can be examined to help prioritize pages for content review or refresh.

These are observed values from the starter dataset, so they support investigating this lane but do not by themselves prove that a refresh will improve performance.

## 3) Quick look at the data

I will load the starter dataset and use a few real numbers to check whether the data contains useful signals for prioritizing pages for content review or refresh.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

**What I can say:**  
This work can identify observed patterns and directional signals in the starter dataset that may help prioritize pages for content review or refresh. The output can be used as decision support for an SEO or content team.

**What I can't say:**  
I cannot claim that the observed patterns are causal, that refreshing a page will definitely improve its performance, or that the analysis can predict or explain Google's ranking algorithm.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.